### DISCUSSION TITLE
Restricting the role for public user create

### DISCUSSION THREAD

**Discussion by krazyjakee** [directus/directus#11229] [Archive]
I see no way to restrict the role when allowing the public to create a new user.

I've tried custom filters `role.id equals x` but then it expects an object from the client and when I send an object it says `user.id not allowed`. I try putting a default value for the role but that throws `you do not have permission`. There seems to be no way to restrict the role.

**Comment by rijkvanzanten** [comment]
I'd recommend removing the `role` field from the Field Permissions (so they can't set the role at all), and then using Presets to pre-fill `role` with the role you want to use for public users

**Reply by krazyjakee** [reply]
When I do that it throws a 403 Forbidden.

When I allow access to the role field and remove the pre-fill role, undoing what you suggested, I'm able to add a user without a problem but of course, without the desired role restrictions.

**Reply by azrikahar** [reply]
The steps outlined by @rijkvanzanten is 100% correct, but I believe you are facing this issue with field presets here https://github.com/directus/directus/issues/10782 which is "modifying" your field presets from:

```js
{
    "role": "c0e94dfb-4143-4c66-832f-ae0a7954561b"
}
```

to

```js
{
    "role": {
        "_eq": "c0e94dfb-4143-4c66-832f-ae0a7954561b"
    }
}
``` 

Which ended up being an unacceptable format, thus the API throws an exception (Forbidden just to avoid leaking underlying error).
